lib/src/webcrypto/webcrypto.ecdh.dart

```diff
@@ -14,43 +14,223 @@
 
 part of 'webcrypto.dart';
 
+/// Elliptic Curve Diffie-Hellman (ECDH) is a key agreement protocol that allows
+/// two parties to establish a shared secret over an insecure channel.
+/// 
+/// An [EcdhPrivateKey] instance represents a private key that can be used to
+/// derive a shared secret with a public key.
+/// 
+/// Instances of [EcdhPrivateKey] can be imported from:
+/// * PKCS8 Key using [EcdhPrivateKey.importPkcs8Key], and,
+/// * JSON Web Key using [EcdhPrivateKey.importJsonWebKey].
+/// 
+/// A random key pair can also be generated using [EcdhPrivateKey.generateKey].
+/// 
+/// **Example**
+/// ```dart
+/// import 'package:webcrypto/webcrypto.dart';
+/// 
+/// Future<void> main() async {
+///   // Generate a new key pair using the P-256 curve.
+///   final keyPair = await EcdhPrivateKey.generateKey(EllipticCurve.p256);
+///   
+///   // Derive 256 bits from the private key using the public key.
+///   final derivedBits = await keyPair.privateKey.deriveBits(256, keyPair.publicKey);
+/// 
+///   // Export the private key as a JSON Web Key.
+///   final exportedPrivateKey = await keyPair.privateKey.exportJsonWebKey();
+/// }
+/// ```
 @sealed
 abstract class EcdhPrivateKey {
   EcdhPrivateKey._(); // keep the constructor private.
 
-  // Note. unsupported on Firefox, see EcdsaPrivateKey.importPkcs8Key
+  /// Import [EcdhPrivateKey] using the PKCS8 format[1].
+  /// The [keyData] parameter is an octet string whose 
+  /// contents are the value of the private key.
+  /// The [curve] parameter specifies the curve to use 
+  /// for the key pair.
+  /// 
+  /// **Example**
+  /// ```dart
+  /// import 'package:webcrypto/webcrypto.dart';
+  /// 
+  /// final pkcs8Key = [ /* ... */ ];
+  /// 
+  /// Future<void> main() async {
+  ///   // Import the private key from PKCS8 format.
+  ///   final privateKey = await EcdhPrivateKey.importPkcs8Key(
+  ///     pkcs8Key,
+  ///     EllipticCurve.p256,
+  ///   );
+  /// }
+  /// ```
+  /// 
+  /// [1]: https://datatracker.ietf.org/doc/html/rfc5208
   static Future<EcdhPrivateKey> importPkcs8Key(
     List<int> keyData,
     EllipticCurve curve,
   ) {
+  // Note. unsupported on Firefox, see EcdsaPrivateKey.importPkcs8Key
     return impl.ecdhPrivateKey_importPkcs8Key(keyData, curve);
   }
 
+  /// Import [EcdhPrivateKey] from a [JSON Web Key][1].
+  /// 
+  /// JSON Web Keys imported using [EcdhPrivateKey.importJsonWebKey] must 
+  /// have `"kty": "EC"` and MUST have the following parameters:
+  /// * `"crv"`: The curve used with the key. This MUST match the curve
+  ///  parameter.
+  /// * `"x"`: The x coordinate for the Elliptic Curve point represented
+  /// as a base64url encoded string. The length of this octet string MUST
```

Comment:
```suggestion
  /// as a [base64Url] encoded string. The length of this octet string MUST
```

We can link to: https://api.dart.dev/stable/3.3.4/dart-convert/base64Url-constant.html